swf/src/types.rs

```diff
@@ -1116,31 +1116,153 @@ pub struct GradientRecord {
 
 #[derive(Debug, PartialEq, Clone)]
 pub struct LineStyle {
-    pub width: Twips,
-    pub color: Color,
-    pub start_cap: LineCapStyle,
-    pub end_cap: LineCapStyle,
-    pub join_style: LineJoinStyle,
-    pub fill_style: Option<FillStyle>,
-    pub allow_scale_x: bool,
-    pub allow_scale_y: bool,
-    pub is_pixel_hinted: bool,
-    pub allow_close: bool,
+    pub(crate) width: Twips,
+    pub(crate) fill_style: FillStyle,
+    pub(crate) flags: LineStyleFlag,
+    pub(crate) miter_limit: Fixed8,
 }
 
 impl LineStyle {
-    pub const fn new_v1(width: Twips, color: Color) -> LineStyle {
-        LineStyle {
-            width,
-            color,
-            start_cap: LineCapStyle::Round,
-            end_cap: LineCapStyle::Round,
-            join_style: LineJoinStyle::Round,
-            fill_style: None,
-            allow_scale_x: false,
-            allow_scale_y: false,
-            is_pixel_hinted: false,
-            allow_close: true,
+    #[inline]
+    pub fn new() -> LineStyle {
+        Default::default()
+    }
+
+    #[inline]
+    pub fn allow_close(&self) -> bool {
+        !self.flags.contains(LineStyleFlag::NO_CLOSE)
+    }
+
+    #[inline]
+    pub fn set_allow_close(mut self, val: bool) -> Self {
```

Comment:
Nitpick: Usually `set` means "borrow self, take parameter, return `()`". This method is a slightly different genre: "take self, take parameter, return self with parameter", which allows jQuery-like method chaining.

We probably should have a separate naming convention for methods that are chained like this. I'm partial to `with`, as in `with_allow_close`, `with_allow_scale_x`, etc. (There's a CLI UI toolkit called `cursive` that does this in various places, but I don't think this is a language-wide convention by any means.)

Reply:
I agree with `with_`, thanks!